test/Test/Toml/Gen.hs

```diff
@@ -332,7 +332,28 @@ Array [Double (-563397.0197456297),Double (-308866.62837749254),Double (-29555.3
 Nested array of AnyValue:
 Array [Array [Text "ACyz38VcLz0hxwdFkHTU6PYK8h8CeaiEpI2xAaiZTKBQ3zC1W717cZY35lk8EAK6pPw3WvwIdNktxIV2LrvFSpU8ee6zkXvpvePitW9aspAeeOCF9Q9ry20y7skFZ2qShi7CSx8888zWIqyc8iBkoLNvq4fONLtuUqSw2SlNee4hDIwrnx5O4RuHW1dQfJcnC34h9S0DlIGYP08qq6QHxO4E0HE74cNmiViGm3xpDC8Ro5D8Y6p0FLSN1ELq9Lwm",Text "HhNv0LKICdlKxN"],Array [Integer 986479839551009895,Integer 8636972066308796678,Integer (-3464941350081979804),Integer (-6560688879547055621),Integer (-4749037439349044738)],Array []]
 -}
-genArray :: MonadGen m => m (Value 'TArray)
+genArray :: (MonadGen m, GenBase m ~ Identity) => m (Value 'TArray)
 genArray = Gen.recursive Gen.choice
     [Gen.choice $ map genArrayFrom noneArrayList]
     [Array <$> Gen.list (Range.constant 0 5) genArray]
+
+-- filters
+
+-- | True if Text is valid based on TOML syntax (https://github.com/toml-lang/toml#string)
+validText :: Text -> Bool
+validText = validString . Text.unpack
+  where
+    validString :: String -> Bool
+    validString [] = True
+    validString ['\\'] = False
+    validString ('\\':'b':s) = validString s
+    validString ('\\':'t':s) = validString s
+    validString ('\\':'n':s) = validString s
+    validString ('\\':'f':s) = validString s
+    validString ('\\':'r':s) = validString s
+    validString ('\\':'"':s) = validString s
+    validString ('\\':'\'':t) = validString t
+    validString ('\\':'\\':t) = validString t
+    validString ('\\':'u':n1:n2:n3:n4:t) = (all Char.isHexDigit [n1, n2, n3, n4]) && validString t
```

Comment:
This function is becoming too complicated... I'm afraid it will require some time to understand what's happening here and make sure that there's no better approach... At this point the previous solution doesn't look that bad :slightly_smiling_face:

Reply:
Bummer, I thought it was pretty slick 😂 Anyway, no problem to revert it